pandas/core/arrays/arrow/array.py

```diff
@@ -521,6 +525,57 @@ def dropna(self: ArrowExtensionArrayT) -> ArrowExtensionArrayT:
         else:
             return type(self)(pc.drop_null(self._data))
 
+    @doc(ExtensionArray.fillna)
+    def fillna(
+        self: ArrowExtensionArrayT,
+        value: object | ArrayLike | None = None,
+        method: FillnaOptions | None = None,
+        limit: int | None = None,
+    ) -> ArrowExtensionArrayT:
+
+        value, method = validate_fillna_kwargs(value, method)
+
+        if is_array_like(value):
+            value = cast(ArrayLike, value)
+            if len(value) != len(self):
+                raise ValueError(
+                    f"Length of 'value' does not match. Got ({len(value)}) "
+                    f" expected {len(self)}"
+                )
+
+        def convert_fill_value(value, pa_type, dtype):
+            if value is None:
+                return value
+            if isinstance(value, (pa.Scalar, pa.Array, pa.ChunkedArray)):
+                return value
+            if is_array_like(value):
+                func = pa.array
+            else:
+                func = pa.scalar
+            try:
+                value = func(value, type=pa_type, from_pandas=True)
+            except pa.ArrowTypeError:
+                raise TypeError(f"Invalid value '{str(value)}' for dtype {dtype}")
+            return value
+
+        fill_value = convert_fill_value(value, self._data.type, self.dtype)
+
+        try:
+            if method is None and limit is None:
+                return type(self)(pc.fill_null(self._data, fill_value=fill_value))
+            elif method == "pad" and limit is None and not pa_version_under7p0:
```

Comment:
Given the use of `pa_version_under7p0`, we want to call `fallback_performancewarning` that a slower path (the super path) is being called